src/crud_ops.rs

```diff
@@ -143,6 +225,51 @@ impl Client {
         counter!(metrics::total_delete_ops).increment(1);
         with_retries!(self, self.delete_impl(path).await)
     }
+
+    async fn bulk_delete_impl(&self, paths: &Vec<Path>) -> crate::Result<Vec<(Path, crate::Error)>> {
+        let stream = stream::iter(paths.iter().map(|path| Ok(path.clone()))).boxed();
+        let indexed_results = self.store.delete_stream(stream)
+            .enumerate()
+            .filter_map(|(index, result)| async move {
+                Some((index, result))
+            }).collect::<Vec<(usize, Result<Path, object_store::Error>)>>().await;
```

Comment:
nit: This `collect` can omit the item type because it can be inferred (the type of collection `Vec` still needs to be explicit):
```suggestion
            }).collect::<Vec<_>>().await;
```